website/static/js/scrollspy.js

```diff
@@ -0,0 +1,37 @@
+const epsilon = 10;
+let timer;
+const onScroll = () => {
+    if (timer) {  // throttle
+        return;
+    }
+    timer = setTimeout(() => {
+        let found = false;
+        const headings = document.querySelectorAll('.toc-headings > li > a');
+        // Some computations based on
+        // https://github.com/makotot/scrollspy/blob/master/src/js/modules/scrollspy.js
+        const scrollTop = document.documentElement.scrollTop ||
+            document.body.scrollTop;
```

Comment:
All browsers except IE<9, which Docusaurus and we don't support, supports `window.pageYOffset` (https://developer.mozilla.org/en-US/docs/Web/API/Window/pageYOffset).